solr-search/src/main/java/edu/unc/lib/dl/search/solr/service/ChildrenCountService.java

```diff
@@ -0,0 +1,228 @@
+        addChildrenCounts(containers, principals, CHILD_COUNT, null);
+    }
+
+    /**
+     * Adds a count of the number of children contained by each object in result
+     * list.
+     *
+     * The count will be calculated based on any restrictions from baseQuery if
+     * it is provided. The count will be stored in each containers count map
+     * under the provided countKey.
+     *
+     * @param containers containers to add child counts to.
+     * @param principals agent principals
+     * @param countKey key of the count to store.
+     * @param baseQuery Optional. Starting query that will be used for
+     *            calculating child counts.
+     */
+    public void addChildrenCounts(List<BriefObjectMetadata> containers, AccessGroupSet principals,
+            String countKey, SolrQuery baseQuery) {
+
+        Assert.notNull(containers);
+        if (containers.size() == 0) {
+            return;
+        }
+
+        log.debug("Adding child counts of type {} to result list", countKey);
+
+        SolrQuery solrQuery;
+        if (baseQuery == null) {
+            // Create a base query since we didn't receive one
+            solrQuery = createBaseQuery(principals);
+        } else {
+            // Starting from a base query
+            solrQuery = baseQuery.getCopy();
+            // Make sure we aren't returning any normal results
+            solrQuery.setRows(0);
+            // Remove all facet fields so we are only getting ancestor path
+            if (solrQuery.getFacetFields() != null) {
+                for (String facetField : solrQuery.getFacetFields()) {
+                    solrQuery.removeFacetField(facetField);
+                }
+            }
+        }
+
+        // Algorithm for getting child counts for list of objects currently involves
+        // retrieving counts for every object in Solr and then matching
+        // the counts back to the metadata object using a binary search. We
+        // should reevaluate this as things scale, but was introduced as more
+        // efficient than hundreds of queries per request.
+        String ancestorPathField = solrField(ANCESTOR_PATH);
+        solrQuery.setFacetMinCount(1);
+        solrQuery.addFacetField(ancestorPathField);
+
+        solrQuery.add("f." + ancestorPathField + ".facet.limit", Integer.toString(Integer.MAX_VALUE));
+        // Sort by value rather than count so that earlier tiers will come first in case the result gets cut off
+        solrQuery.setFacetSort("index");
+
+        try {
+            QueryResponse queryResponse = this.executeQuery(solrQuery);
+            assignChildrenCounts(queryResponse.getFacetField(ancestorPathField), containers, countKey);
+        } catch (SolrServerException e) {
+            throw new SolrRuntimeException(e);
+        }
+    }
+
+    private SolrQuery createBaseQuery(AccessGroupSet principals) {
+        SolrQuery solrQuery = new SolrQuery();
```

Comment:
since lines 160-169 are the same as lines 61-70, can you create a private method to set up the query?

Reply:
Oops, `createBaseQuery` is actually already the private method that was supposed to replace 61-70, and is being used at line 61, but i forgot to delete the redundant lines